### PR TITLE
[cmake] More TARGET usage (opengl/opengles/iconv/Pulseaudio)

### PR DESCRIPTION
**Pull request by fuzzard**
## Description
Migrate more modules to full TARGET usage

## Motivation and context
Migrate OpenGL/OpenGLES/ICONV/PulseAudio to full TARGET usage.
View without whitespace (append `?w=1` to the URL) for easier review, but same concept as other TARGET migrations. minimal functional change to the find.

The one change that does affect functional, ios/tvos do have GLES3 headers available. Ive added their search to the gles3_includes. They will now find glesv3, so the system_gl.h had to be altered, as Apple's gles3 headers include the gles2 defines/functions, so we can only link one or the other for apple.

OpenGL find's are just unified (same as opengles). should be no need to separate, but homebrew stuff may bite me in the ass (as always). I'll deal with that if it becomes an issue.

## How has this been tested?
Configure, lets go jenkins!

## What is the effect on users?
N/A

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Review by garbear — APPROVED** [review]
Changes look like they were done correctly

File: cmake/modules/FindOpenGLES.cmake [hunk 1]
```diff
@@ -1,50 +1,62 @@
...
-  set(OPENGLES_LIBRARIES ${OPENGLES_gl_LIBRARY})
-  if(OPENGLES3_INCLUDE_DIR)
-    set(OPENGLES_INCLUDE_DIRS ${OPENGLES_INCLUDE_DIR} ${OPENGLES3_INCLUDE_DIR})
-    set(OPENGLES_DEFINITIONS -DHAS_GLES=3)
-    mark_as_advanced(OPENGLES_INCLUDE_DIR OPENGLES3_INCLUDE_DIR OPENGLES_gl_LIBRARY)
-  else()
-    set(OPENGLES_INCLUDE_DIRS ${OPENGLES_INCLUDE_DIR})
-    set(OPENGLES_DEFINITIONS -DHAS_GLES=2)
-    mark_as_advanced(OPENGLES_INCLUDE_DIR OPENGLES_gl_LIBRARY)
+    set_property(GLOBAL APPEND PROPERTY INTERNAL_DEPS_PROP OpenGL::GLES)
```
Comment (vpeter4) [inline cmake/modules/FindOpenGLES.cmake:60]: @fuzzard: Is here missing line `set(OPENGLES_LIBRARIES ${OPENGLES_gl_LIBRARY})`?
Without it I get absolute path from build host in kodi:
```
CoreELEC VIM4:~ # ldd /usr/lib/kodi/kodi.bin
        /home/docker/build_dir/CoreELEC-21/build.CoreELEC-Amlogic-ne.aarch64-21/toolchain/aarch64-libreelec-linux-gnu/sysroot/usr/lib/libGLESv2.so => not found
```

Comment (fuzzard) [reply cmake/modules/FindOpenGLES.cmake:60]: Short answer is no, that isnt a correct fix. the *_LIBRARIES variables (and *_DEFINITIONS, *_INCLUDE_DIRS) are a legacy way to pass info around the cmake world in our macros. Populating the target should provide all the information, of which cmake then extracts/uses as it needs behind the scenes.

At a pure blind guess im guessing some rpath bullshit, but no idea at this point in time. Grab a CMakeCache.txt from the kodi package build and we'll see whats in it

Comment (vpeter4) [reply cmake/modules/FindOpenGLES.cmake:60]: CMakeCache.txt is 100% the same
https://gist.github.com/vpeter4/c7bfbdded5674aab37e97493d7e7763c

but in build.ninja I see that it links GLESv2 library 3 times when ok and only 1 time when not ok
diff at "Link the executable kodi.bin" block
ok https://gist.github.com/vpeter4/df53b76f8a6186f7b5dc7fa01fdb97a1
not https://gist.github.com/vpeter4/955406218b2dea5535aa3b44a5644ac3

Comment (fuzzard) [reply cmake/modules/FindOpenGLES.cmake:60]: Its not a linker failure you're experiencing though, so the number of times it appears on the linker line for this particular issue isnt relevant as it succesfully links. What i believe its not doing is the change of rpath for the isntall step.

Cmake when it links for the build will extend out the full path of libs for that step to be able to execute from the build dir, but when it does an install its meant to filter out things like the rpath to things potentially provided for CMAKE_INSTALL_RPATH and friends.
Im not sure if its definitely the install rpath thats the issue, or potentially that theres no IMPORTED_SONAME or maybe an INSTALL_RPATH property. I havent come across a need for them as yet, so im grabbing at straws here.

Unfortunately at this point ive been unable to build libreelec as im an aarch64 host, and i get build failures from the internal glib in pkg-config. 

```
{standard input}: Assembler messages:
{standard input}:673: Error: selected processor does not support `eor3 v0.16b,v0.16b,v1.16b,v3.16b'
```

So im trying to find a reproducer without having to build the elec's at this point.

Comment (fuzzard) [reply cmake/modules/FindOpenGLES.cmake:60]: No, it wont be that. As mentioned, cmake for the build step will use full lib paths. This is normal. When a target is installed, cmake relinks and updates/changes rpaths and other things to be installed.

Comment (vpeter4) [reply cmake/modules/FindOpenGLES.cmake:60]: I think this issue comes from the fact that library `libGLESv2.so` doesn't have `SONAME` set (`objdump -p` doesn't show it).
There are 2 options:

1. set `SONAME` using `patchelf` binary
2. modifying cmake file
```
--- a/cmake/modules/FindOpenGLES.cmake	2023-10-07 20:07:55.000000000 +0200
+++ b/cmake/modules/FindOpenGLES.cmake	2023-10-11 15:08:10.933170795 +0200
@@ -42,9 +42,10 @@ if(NOT TARGET OpenGL::GLES)
       endif()
     endif()
 
-    add_library(OpenGL::GLES UNKNOWN IMPORTED)
+    add_library(OpenGL::GLES SHARED IMPORTED)
     set_target_properties(OpenGL::GLES PROPERTIES
                                        IMPORTED_LOCATION "${OPENGLES_gl_LIBRARY}"
+                                       IMPORTED_NO_SONAME TRUE
                                        INTERFACE_INCLUDE_DIRECTORIES "${OPENGLES_INCLUDE_DIR}")
```

I would rather see option 2 but probably needs to cover shared and static libraries.

Comment (fuzzard) [reply cmake/modules/FindOpenGLES.cmake:60]: Definitely can be done in cmake. Can it be left as an unknown imported but just add soname property? As far as I read we might be able to add for everything, as platforms that don't use soname just disregard the propery

**Pull request merged**